x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/Security.java

```diff
@@ -628,6 +636,61 @@ protected Clock getClock() {
         return components;
     }
 
+    private void generateElasticPassword(SecurityIndexManager.State previousState, SecurityIndexManager.State currentState) {
```

Comment:
I'd prefer this wasn't in `Security`.
We have a bad habit of just adding more and more into this class, and it's unnecessary. 

Can we have a `GenerateInitElasticPasswordListener`?

Reply:
Sure thing, I will amend

Reply:
Done in 79a6c21